datatransfer/impl/graphsync/graphsync.go

```diff
@@ -116,10 +135,75 @@ func (impl *graphsyncImpl) InProgressChannels() map[datatransfer.ChannelID]datat
 }
 
 // TODO: implement for https://github.com/filecoin-project/go-data-transfer/issues/14
+// ReceiveRequest takes an incoming request, validates the voucher and processes the message.
 func (receiver *graphsyncReceiver) ReceiveRequest(
 	ctx context.Context,
 	sender peer.ID,
 	incoming message.DataTransferRequest) {
+
+	voucher, err := receiver.validateVoucher(sender, incoming)
+	if err != nil {
```

Comment:
Since this PR is only about implementing validation, these printfs are placeholders for the next issue that handles the results.